src/events/kyuubi.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Limited
+# See LICENSE file for licensing details.
+
+"""Kyuubi related event handlers."""
+
+import ops
+from ops import CharmBase
+
+from constants import KYUUBI_CLIENT_RELATION_NAME
+from core.context import Context
+from core.workload import KyuubiWorkloadBase
+from events.base import BaseEventHandler, compute_status
+from managers.kyuubi import KyuubiManager
+from providers import KyuubiClientProvider
+from utils.logging import WithLogging
+
+
+class KyuubiEvents(BaseEventHandler, WithLogging):
+    """Class implementing Kyuubih related event hooks."""
+
+    def __init__(self, charm: CharmBase, context: Context, workload: KyuubiWorkloadBase):
+        super().__init__(charm, "kyuubi")
+
+        self.charm = charm
+        self.context = context
+        self.workload = workload
+
+        self.kyuubi = KyuubiManager(self.workload)
+        self.kyuubi_client = KyuubiClientProvider(self.charm, KYUUBI_CLIENT_RELATION_NAME)
+
+        self.framework.observe(self.charm.on.install, self._on_install)
+        self.framework.observe(self.charm.on.kyuubi_pebble_ready, self._on_kyuubi_pebble_ready)
+        self.framework.observe(self.charm.on.update_status, self._update_event)
+        self.framework.observe(self.charm.on.config_changed, self._on_config_changed)
+
+    @compute_status
+    def _on_install(self, event: ops.InstallEvent) -> None:
+        """Handle the `on_install` event."""
+
+    @compute_status
+    def _on_config_changed(self, event: ops.ConfigChangedEvent) -> None:
+        """Handle the on_config_changed event."""
+        if not self.charm.unit.is_leader():
+            return
+
+        if not self.workload.ready():
+            event.defer()
+            return
+
+        self.kyuubi.update(
+            s3_info=self.context.s3,
+            metastore_db_info=self.context.metastore_db,
+            auth_db_info=self.context.auth_db,
+            service_account_info=self.context.service_account,
+        )
+
+    @compute_status
+    def _update_event(self, event):
+        """Handle the update event hook."""
```

Comment:
**niptick** I would just add the `pass` here.

Reply:
Updated.